vllm/distributed/kv_transfer/kv_connector/v1/base.py

```diff
@@ -195,6 +195,13 @@ def get_finished(
     # Scheduler-side methods
     # ==============================
 
+    def should_free_pending_on_abort(self) -> bool:
```

Comment:
Discussed offline but for more flexibility and for proper multi-connector support, it would be good to add `req_id: str` arg here.

Reply:
@robertgshaw2-redhat we still might have an annoying edge case here for multi-connector, since we could still receive a finished event from the worker side of a connector which we also free here. So we might think two connectors were finished with the request in that case when really one is still transferring the blocks.

Since it's just counts that we propagate back from the worker side to the scheduler, we also don't have a way of keeping track of the completions on a per-connector basis.